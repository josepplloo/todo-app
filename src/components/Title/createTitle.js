import constants from './constants';

import './title.scss';

function createTitle() {
  const titleElement = document.createElement('h1');
  titleElement.setAttribute('label', constants.altText);
  titleElement.innerHTML = constants.titleText;
  titleElement.className = 'app-title';
  return titleElement;
}

export default createTitle;
