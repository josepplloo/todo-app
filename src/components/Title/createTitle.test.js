import createTitle from './createTitle';

it('should return an <h1 /> element', () => {
  const titleElement = createTitle();
  expect(titleElement.tagName).toBe('H1');
});

it('should have "todo" as title value', () => {
  const titleElement = createTitle();
  expect(titleElement.innerHTML).toBe('todos');
});
it('should have a styles and layout like the snapshot', () => {
  const titleElement = createTitle();
  expect(titleElement).toMatchSnapshot();
});
