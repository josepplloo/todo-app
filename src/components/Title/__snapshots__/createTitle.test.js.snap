// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have a styles and layout like the snapshot 1`] = `
<h1
  class="app-title"
  label="to do list"
>
  todos
</h1>
`;
